File: gptcomet/llms/xai.py
from typing import Any

from .openai import OpenaiLLM


class XaiLLM(OpenaiLLM):
    """XAI (讯飞星火) LLM provider implementation."""

    def __init__(self, config: dict[str, Any]):
        super().__init__(config)

        self.api_base = config.get("api_base") or "https://api.x.ai/v1/"
        self.model = config.get("model") or "grok-beta"

    @classmethod
    def get_required_config(cls) -> dict[str, tuple[str, str]]:
        return {
            "api_base": ("https://api.x.ai/v1/", "Enter XAI API Base URL"),
            "model": ("grok-beta", "Enter model name"),
            "api_key": ("", "Enter API key"),
            "max_tokens": ("1024", "Enter max tokens"),
        }

File: internal/llm/xai.go
package llm

import (
	"github.com/belingud/gptcomet/pkg/config"
	"github.com/belingud/gptcomet/pkg/types"
)

const (
	DefaultXAIAPIBase = "https://api.x.ai/v1"
	DefaultXAIModel   = "grok-beta"
)

// XAILLM implements the LLM interface for XAI
type XAILLM struct {
	*OpenAILLM
}

// NewXAILLM creates a new XAILLM
func NewXAILLM(config *types.ClientConfig) *XAILLM {
	BuildStandardConfigSimple(config, DefaultXAIAPIBase, DefaultXAIModel)

	return &XAILLM{
		OpenAILLM: NewOpenAILLM(config),
	}
}

func (x *XAILLM) Name() string {
	return "xai"
}

// GetRequiredConfig returns provider-specific configuration requirements
func (x *XAILLM) GetRequiredConfig() map[string]config.ConfigRequirement {
	return map[string]config.ConfigRequirement{
		"api_base": {
			DefaultValue:  DefaultXAIAPIBase,
			PromptMessage: "Enter XAI API base",
		},
		"api_key": {
			DefaultValue:  "",
			PromptMessage: "Enter API key",
		},
		"model": {
			DefaultValue:  DefaultXAIModel,
			PromptMessage: "Enter model name",
		},
		"max_tokens": {
			DefaultValue:  "1024",
			PromptMessage: "Enter max tokens",
		},
	}
}
